### ISSUE TITLE
[Link to pay missing] user recieves an alert containing a link to pay but link is missing/empty.

### ISSUE THREAD

**Issue by ghost** [btcpayserver/xenforo#10]
Hello. I am 2 issues that i think are related. 
I installed the btcpay add-on to my xenforo forum. Followed the instructions. 
When i purchase an upgrade. I receive an alert with a link but the link is missing/empty.

When i log into the btcpay instance i can see the invoice was generated but on xenforo there is no link to pay.
When i visit:
`https://mywebsite.com/forum/payment_callback.php?_xfProvider=btcPayServer`
I see a message "Invalid payment profile."

Is it possible it's not working because i installed my forum inside a subfolder? I'm not sure but here is my forum's information.

Info:
**Xenforo  v2.3.3**
PHP version - [8.3.11]
MySQL version - 10.6.19 (10.6.19-MariaDB-cll-lve)
PHP memory_limit - 512M
PHP post_max_size - 300M 
PHP upload_max_filesize - 512M
PHP max_input_vars - 1000
PHP max_execution_time - 30
cURL version - 7.87.0
cURL SSL version - OpenSSL/1.1.1w
OpenSSL version - OpenSSL 1.1.1w 11 Sep 2023 
Snuffleupagus enabled - No
Imagick support - Yes
EXIF support - Yes
GZip support - Yes
gmp support - No
ZipArchive support Yes

Here is some pictures:

Add-on installed
![btcpay10](https://github.com/user-attachments/assets/e9ca76a9-aaaa-427c-8abb-9e7421363551)

Add-on settings 
![btc1](https://github.com/user-attachments/assets/378170ff-f90b-48ae-89cb-1e6ac41c34d1)

User upgrade setting
![btcpay8](https://github.com/user-attachments/assets/09ad5aa9-bce1-4dd4-b4d1-479d34c1b89c)

Payment profile
![btcpay9](https://github.com/user-attachments/assets/e06fab41-95e7-49c9-86ad-80d821f2990d)

Webhooks settings
![btcpay3](https://github.com/user-attachments/assets/7fd5cc9c-aa5c-4d2b-8c37-7bc85d25f406)

Api settings. I added only the api that the instructions asked for but didn't work so afterwards i installed all the apis for testing but still doesn't work.
![btcpay2](https://github.com/user-attachments/assets/8954043f-ca61-4b81-bf49-6efcc0423203)

Payload URL
![btcpay7](https://github.com/user-attachments/assets/5c4d6c2b-a1ea-4479-9b8d-42d85104fc0a)

Account upgrade
![btcpay6](https://github.com/user-attachments/assets/029976fa-c8a1-42b3-b85e-2536c2339a7f)

Invoice Alert. - when i click on the link nothing happens. When i copy the link. It does not copy because the link is missing. Link is empty.
![btcpay5](https://github.com/user-attachments/assets/6529397d-267e-47cd-90c0-7ded54598a1d)
```
When i right click inspect the link i see this:
<a data-template-name="public:alert_user_btcpayprovider_invoice_created" id="_xfUid-1-1728575018" data-xf-click="">The invoice</a>
#_xfUid-1-1728575018
html#XF.has-js.template-forum_list.has-no-touchevents.has-passiveeventlisteners.has-hiddenscroll.has-overflowanchor.has-no-displaymodestandalone.has-flexgap.has-os-linux.has-browser-mozilla.has-pointer-nav body div#js-XFUniqueId61.menu.menu--structural.menu--medium.menu--potentialFixed.menu--right.is-active.is-complete div.menu-content div.js-alertsMenuBody div.menu-scroller ol.listPlain li.alert.js-alert.menu-row.menu-row--separated.menu-row--clickable div.fauxBlockLink div.contentRow div.contentRow-main.contentRow-main--close a#_xfUid-1-1728575018
//*[@id="_xfUid-1-1728575018"]
```

Invoice is present in my btcpay store
![btcpay4](https://github.com/user-attachments/assets/62cea661-a00e-4b9c-a091-38897f167d22)

**Comment by 021-projects** [comment]
Hi!
The link should be missing because it is processed using JS. It is also normal that you see the message "Invalid payment profile" when you go directly to the webhook link. I'm not quite clear what code you sent below the link, HTML can't look like that, so it would be great to see it as a screenshot to better understand what you mean. And please check your console for errors when you try to open an invoice from the alert. I also want to ask if there are any errors in your log?

**Comment by git-3000** [comment]
Where can the user find the invoice so he can pay? 

The xenforo user does not have a way to pay the invoice because there is no invoice link.

There is no errors in the administration panel. Is there somewhere else i can check to find errors?

**Comment by 021-projects** [comment]
Are there any errors in the browser console when clicking on the invoice link? Does a payment popup open for you when you select BTCPay as your payment method and click pay?

**Comment by mar101xy** [comment]
I upgraded my BTCPay Server. I thought maybe that would solve the issue but it didn't.
v2.0.2+b70c004f2

My PHP 8.1.30

No payment popup pop's up. 

I looked at the browser console and i see this error:

```
Cookie warnings 2
Cookie “_ga” does not have a proper “SameSite” attribute value. Soon, cookies without the “SameSite” attribute or with an invalid value will be treated as “Lax”. This means that the cookie will no longer be sent in third-party contexts. If your application depends on this cookie being available in such contexts, please add the “SameSite=None“ attribute to it. To know more about the “SameSite“ attribute, read https://developer.mozilla.org/docs/Web/HTTP/Headers/Set-Cookie/SameSite index.php
Cookie “_gid” does not have a proper “SameSite” attribute value. Soon, cookies without the “SameSite” attribute or with an invalid value will be treated as “Lax”. This means that the cookie will no longer be sent in third-party contexts. If your application depends on this cookie being available in such contexts, please add the “SameSite=None“ attribute to it. To know more about the “SameSite“ attribute, read https://developer.mozilla.org/docs/Web/HTTP/Headers/Set-Cookie/SameSite index.php
Feature Policy: Skipping unsupported feature name “clipboard-read”. btcpay.js:45:12
Feature Policy: Skipping unsupported feature name “clipboard-write”.
```

Im going to try the btc demo site and see if that solves the issue. Maybe the problem is with my btcpay?

**Comment by 021-projects** [comment]
> `The loading of “https://btcpay.mar101xy.com/invoice?id=P7VVJ9mL6MRvkkHvTSkGmn&view=modal” in a frame is denied by “X-Frame-Options“ directive set to “deny“.`

So your server sends a header that prohibits displaying BtcPay content in Iframe. Check your server configs (Nginx or Apache) for the presence of the line with "X-Frame-Options".  Remove this line

**Comment by 021-projects** [comment]
> I want to request that when i press on purchase the invoice should pop-up or a new window browser should open and take the user to the btcpay website to pay the invoice. I think the user might press \"purchase\" many times thinking it won't work

This is how it should work. When you click the purchase button, the payment popup should open. Do you have any errors when you click this button? Have you tried waiting longer?

**Comment by mar101xy** [comment]
In the inspect console i see lots of errors. I used chatgpt and it seems to think its a virtual host file error. Something to do with websocket and not your add-on. I'll keep trying different things and see if i can get it working the correct way

```
GET
wss://btcpay.example.com/invoice/status/ws?invoiceId=REDACTED
[HTTP/1.1 404 Not Found 348ms]

GET
    wss://btcpay.example.com/invoice/status/ws?invoiceId=REDACTED
Status
404
Not Found
VersionHTTP/1.1
Transferred415 B (0 B size)
DNS ResolutionSystem

    Connection
        Keep-Alive
    Content-Length
        0
    Content-Security-Policy
        frame-ancestors 'self' https://example.com
    Content-Security-Policy
        script-src 'self'
    Date
        Mon, 18 Nov 2024 00:29:44 GMT
    Keep-Alive
        timeout=5, max=100
    Referrer-Policy
        no-referrer-when-downgrade
    Referrer-Policy
        same-origin
    Server
        Kestrel
    X-Content-Type-Options
        nosniff
    X-Content-Type-Options
        nosniff
    X-Frame-Options
        DENY
    X-XSS-Protection
        1; mode=block
    X-XSS-Protection
        1; mode=block

    Accept
        */*
    Accept-Encoding
        gzip, deflate, br, zstd
    Accept-Language
        en-CA,en-US;q=0.7,en;q=0.3
    Cache-Control
        no-cache
    Connection
        keep-alive, Upgrade
    Cookie
        [REDACTED]
    DNT
        1
    Host
        btcpay.example.com
    Origin
        https://btcpay.example.com
    Pragma
        no-cache
    Sec-Fetch-Dest
        empty
    Sec-Fetch-Mode
        websocket
    Sec-Fetch-Site
        same-origin
    Sec-WebSocket-Extensions
        permessage-deflate
    Sec-WebSocket-Key
        [REDACTED]
    Sec-WebSocket-Version
        13
    Upgrade
        websocket
    User-Agent
        Mozilla/5.0 (X11; Linux x86_64; rv:132.0) Gecko/20100101 Firefox/132.0

Firefox can’t establish a connection to the server at wss://btcpay.example.com/invoice/status/ws?invoiceId=REDACTED. checkout.js:288:34

Error while connecting to websocket for invoice notifications (callback): 
error { target: WebSocket, isTrusted: true, srcElement: WebSocket, currentTarget: WebSocket, eventPhase: 2, bubbles: false, cancelable: false, returnValue: true, defaultPrevented: false, composed: false, … }
checkout.js:293:37
    onerror https://btcpay.example.com/checkout/checkout.js?v=REDACTED:293
    (Async: EventHandlerNonNull)
    listenIn https://btcpay.example.com/checkout/checkout.js?v=REDACTED:292
    mounted https://btcpay.example.com/checkout/checkout.js?v=REDACTED:234
    VueJS 7
    initApp https://btcpay.example.com/checkout/checkout.js?v=REDACTED:85
    e https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    i https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    ...
    <truncated for brevity>
Firefox can’t establish a connection to the server at wss://btcpay.example.com/invoice/status/ws?invoiceId=REDACTED. checkout.js:288:34

Error while connecting to websocket for invoice notifications (callback): 
error { target: WebSocket, isTrusted: true, srcElement: WebSocket, currentTarget: WebSocket, eventPhase: 2, bubbles: false, cancelable: false, returnValue: true, defaultPrevented: false, composed: false, … }
checkout.js:293:37
    onerror https://btcpay.example.com/checkout/checkout.js?v=REDACTED:293
    (Async: EventHandlerNonNull)
    listenIn https://btcpay.example.com/checkout/checkout.js?v=REDACTED:292
    mounted https://btcpay.example.com/checkout/checkout.js?v=REDACTED:234
    VueJS 7
    initApp https://btcpay.example.com/checkout/checkout.js?v=REDACTED:85
    e https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    i https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    i https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    value https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    value https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    value https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    value https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    value https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    value https://btcpay.example.com/vendor/i18next/i18nextHttpBackend.min.js?v=REDACTED:1
    O https://btcpay.example.com/vendor/i18next/i18nextHttpBackend.min.js?v=REDACTED:1
    (Async: promise callback)
    O https://btcpay.example.com/vendor/i18next/i18nextHttpBackend.min.js?v=REDACTED:1
    (Async: promise callback)
    O https://btcpay.example.com/vendor/i18next/i18nextHttpBackend.min.js?v=REDACTED:1
    default https://btcpay.example.com/vendor/i18next/i18nextHttpBackend.min.js?v=REDACTED:1
    value https://btcpay.example.com/vendor/i18next/i18nextHttpBackend.min.js?v=REDACTED:1
    value https://btcpay.example.com/vendor/i18next/i18nextHttpBackend.min.js?v=REDACTED:1
    (Async: promise callback)
    value https://btcpay.example.com/vendor/i18next/i18nextHttpBackend.min.js?v=REDACTED:1
    value https://btcpay.example.com/vendor/i18next/i18nextHttpBackend.min.js?v=REDACTED:1
    value https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    value https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    value https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    value https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    value https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    value https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    i https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    value https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    f https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    (Async: setTimeout handler)
    value https://btcpay.example.com/vendor/i18next/i18next.min.js?v=REDACTED:1
    <anonymous> https://btcpay.example.com/checkout/checkout.js?v=REDACTED:467
Source map error: request failed with status 404
Resource URL: https://btcpay.example.com/vendor/dom-confetti/dom-confetti.min.js?v=QfuoqzCN7xyhMNkuHmGC7QDPcb9b6KEmJE1c2D-3W-g
Source Map URL: /sm/b04c41c9e0ca7d063aa88138fff144a5dd6481a3809187211290de788b6184f8.map
```

**Comment by mar101xy** [comment]
I fixed my second issue. 

When i downloaded the add-on. I download the add-on that was released on sept 16 - v1.1.0 thinking that was the latest one. I realized that, that one is for btcpay v1

I downloaded the one below and it was version 2.0.0 for btcpay 2

I upgraded to that version and now it pops up and works without issue. 

thank you!

**Comment by 021-projects** [comment]
> I fixed my second issue. 
> 
> When i downloaded the add-on. I download the add-on that was released on sept 16 - v1.1.0 thinking that was the latest one. I realized that, that one is for btcpay v1
> 
> I downloaded the one below and it was version 2.0.0 for btcpay 2
> 
> I upgraded to that version and now it pops up and works without issue. 
> 
> thank you!

Hi! This is not exactly true. Version 1.x is for XenForo 2.2, and version 2.x is for XenForo 2.3. But I'm glad your problem is solved! Enjoy using it!
